### PR TITLE
Issue #10910: Removed curly braces usage for javadoc tags in translations

### LINKED ISSUE #10910: Remove curly braces usage for javadoc tags in translations
Issue come from https://github.com/checkstyle/checkstyle/pull/10854#discussion_r736815881_

There are failures in tests when autoescape for single quotes added.

I suggest to remove usages of curly braces in messages. At the moment there are only 2 messages (in english + entries in other langs)
```
 grep -R --include "messages.properties" "'{'" .
./src/main/resources/com/puppycrawl/tools/checkstyle/checks/annotation/messages.properties:annotation.missing.override=Must include @java.lang.Override annotation when '{'@inheritDoc'}' Javadoc tag exists.
./src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages.properties:javadoc.invalidInheritDoc=Invalid use of the '{'@inheritDoc'}' tag.
```

So we can change it to 
```
Must include @java.lang.Override annotation when '@inheritDoc' Javadoc tag exists.
```
and 
```
Invalid use of the '@inheritDoc' tag.
```

Other option is to disable escape and manage escapes by yourselves (user has to add escapes in crowdin), but I think it is not a very convinient way.

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #10910: Remove curly braces usage for javadoc tags in translations

### CODE REVIEW

**Comment by rnveach** [comment]
This is not a minor change if it resolves an issue.

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/annotation/messages_de.properties [hunk 1]
```diff
@@ -2,7 +2,7 @@ annotation.incorrect.style=Die Annotation muss dem Stil ''{0}'' folgen.
 annotation.location=Die Annotation ''{0}'' hat eine unerwartete Einrückungstiefe von {1} (erwartet: {2}).
 annotation.location.alone=Die Annotation ''{0}'' sollte in einer eigenen Zeile stehen.
 annotation.missing.deprecated=Eine Deprecated-Annotation (@java.lang.Deprecated) und ein Deprecated-Javadoc-Tag (@deprecated) mit Beschreibung müssen immer zusammen verwendet werden.
-annotation.missing.override=Wenn das '{'@inheritDoc'}'-Javadoc-Tag verwendet wird, muss eine Override-Annotation (@java.lang.Override) vorhanden sein.
+annotation.missing.override=Wenn das '@inheritDoc'-Javadoc-Tag verwendet wird, muss eine Override-Annotation (@java.lang.Override) vorhanden sein.
```
Comment (strkkk) [inline src/main/resources/com/puppycrawl/tools/checkstyle/checks/annotation/messages_de.properties:5]: here and in other translations below quote is not escaped

Comment (Vyom-Yadav) [reply src/main/resources/com/puppycrawl/tools/checkstyle/checks/annotation/messages_de.properties]: Fixed it.

**Review by strkkk — APPROVED** [review]

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages_fi.properties [hunk 2]
```diff
@@ -9,7 +9,7 @@ javadoc.empty=Javadoc on tyhjä §.
 javadoc.expectedTag=Javadoc-tagi puuttuu: {0} tagi ''{1}''.
 javadoc.extraHtml=Extra HTML-koodi löytyy: {0}
 javadoc.incompleteTag=Puutteelliset HTML-koodi löytyy: {0}
-javadoc.invalidInheritDoc=Virheellinen käyttö '{'inheritDoc '}' tag.
+javadoc.invalidInheritDoc=Virheellinen käyttö ''inheritDoc'' tag.
```
Comment (rnveach) [inline src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages_fi.properties:12]: If we are going to sync all these tags, then I think they should be consistent. This just has the tag name while others have the `@`.

**Review by rnveach — CHANGES_REQUESTED** [review]
Please make sure this is still up to date. I made changes to the property file that I believe brought one in that you are changing.

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages_fi.properties (hunk 2 above)
Comment (Vyom-Yadav) [reply src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages_fi.properties]: Fixed it, also cross-checked the changes with https://github.com/checkstyle/checkstyle/commit/50675ca340f9d7cc1a55bd2d12108a8b6c35e078

**Comment by romani** [comment]
Still conflict

**Comment by Vyom-Yadav** [comment]
> Still conflict

Resolved now

**Review by rnveach — APPROVED** [review]

**Pull request merged**